### PR TITLE
fix: make interpolate() fill null values in input

### PR DESCRIPTION
**Pull request by wolffcm**
This is the final part of #7287, and requires #7478 to be merged.

This updates how `interpolate()` is implemented to consider null values as missing and to fill them in as other gaps.

### CODE REVIEW

File: iox_query/src/exec/gapfill/algo/interpolate.rs [hunk 1]
```diff
@@ -375,8 +382,8 @@ mod test {
         - "| 1970-01-01T00:00:00.000001200Z | 133  |"
         - "| 1970-01-01T00:00:00.000001300Z | 166  |"
         - "| 1970-01-01T00:00:00.000001400Z | 200  |"
-        - "| 1970-01-01T00:00:00.000001500Z |      |"
-        - "| 1970-01-01T00:00:00.000001600Z |      |"
+        - "| 1970-01-01T00:00:00.000001500Z | 466  |"
+        - "| 1970-01-01T00:00:00.000001600Z | 733  |"
         - "| 1970-01-01T00:00:00.000001700Z | 1000 |"
```
Comment (wolffcm) [inline iox_query/src/exec/gapfill/algo/interpolate.rs:387]: @stuartcarnie This test case illustrates the old behavior of `FillStrategy::LinearInterpolate` and the fixed behavior. A null value in `a0` prevented interpolation for this case.

**Review by alamb — APPROVED** [review]
> This ... requires https://github.com/influxdata/influxdb_iox/pull/7478 to be merged.
Looks like an improvement to me. Thanks @wolffcm 

~I don't understand why this requires https://github.com/influxdata/influxdb_iox/pull/7478 given all the tests pass. Maybe there is some additional end to end test that is required to cover the interaction?~

Update: I see this PR targets a different branch, not `main` 👍

**Comment by wolffcm** [comment]
> Update: I see this PR targets a different branch

Yes, you're right, I should have called that out.

**Comment by wolffcm** [comment]
Fixes #7287.

**Pull request merged**